### ISSUE TITLE
Spike how we rearrange components for Clientside MVT

### ISSUE THREAD

**Issue by andrewscfc** [bbc/simorgh#9872]
**Is your feature request related to a problem? Please describe.**
For our upcoming experiment around recommendations we have a need to change the position and types of components for different variants. We have a problem though where if we cause Visual Journalism includes to re-render they will likely not load correctly on re-render; full background can be found here: https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/architecture-decision-records/2020-08-27-story-page-restricted-clientside-rendering.md We need to identify a technical solution for rearranging blocks once the variant is loaded via [`useOptimizelyVariation`](https://github.com/bbc/simorgh/blob/latest/src/app/hooks/useOptimizelyVariation/index.jsx) that does not cause includes to re-render.

**Describe the solution you'd like**
- Identify a Story fixture page locally that includes an include and verify it loads correctly after making no changes
- Try different methods to rearrange the blocks using the [`useOptimizelyVariation`](https://github.com/bbc/simorgh/blob/latest/src/app/hooks/useOptimizelyVariation/index.jsx) hook and verify the includes block doesn't re-render
- Some ideas to try:
  - Pass in the variation here in ther server component: https://github.com/bbc/simorgh/blob/9d87b8934b5e40f46c2705afcf90001e4e0b3cb5/src/server/index.jsx#L145
    - The logic here can use the variant to insert/move/delete blocks as needed based on the variant: https://github.com/bbc/simorgh/blob/cdf5c11f3087f87a57f95afc8fefeb3e53c4e3ff/src/app/routes/cpsAsset/getInitialData/index.js#L67
    - This in theory would trigger a re-render of the associated components but potentially not the includes if the re-render is intelligent enough
    - This way could be 'purest' but is complex and probably is unlikely to prevent includes from re-rendering
   - Create placeholder components
     -  Add logic [here](https://github.com/bbc/simorgh/blob/cdf5c11f3087f87a57f95afc8fefeb3e53c4e3ff/src/app/routes/cpsAsset/getInitialData/index.js#L67) to add blocks for all possible variants initially
     - Add logic into each block type required to show or hide the content based on the variant returned from the [`useOptimizelyVariation`](https://github.com/bbc/simorgh/blob/latest/src/app/hooks/useOptimizelyVariation/index.jsx) hook
     - This hopefully will cause the relevant component to appear/disappear but not cause the whole article content to re-render
     - This is arguably less maintainable as the variation logic is littered around the codebase and the pattern would need to be repeated each time we did an experiment
     - It is though less likely to cause includes to re-render

There may be other ideas, these are two that come to mind.

**Describe alternatives you've considered**

**Testing notes**
Please test the arrangements of components not the events fired by optimisely. 

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

**Comment by amoore108** [comment]
Had a quick look at this and I think it'll have to be some combination of the 2 suggested approaches, whereby we modified the `pageData` in the `getInitialData` function and then add logic into the `CpsRecommendations` component to figure out which variation to display.

It's tricky to find a solution that we could universally use for future experiments, as each one will be somewhat unique in the approach (I assume?). The trick with this one is the splitting of the recommendations. The other variations are pretty straightforward to implement, as they are either switching out to a different component (requiring some data manipulation in the component) or hiding it completely. The other variation is within the Related Content section, so again a data manipulation task within the component.

It could be useful to have a function in `getInitialData` that adds a block where we want, for example something like:
```
  insertExperimentalBlock({
    shouldInsert: true,
    insertAtIndex: 10,
    blockModel: {
      type: 'wsoj',
      model: {
        type: 'recommendations',
        path: '/api/recommend?recSys=2&limit=2&assetUri=/hindi/india-60392284'
      },
    },
  }),
```

then

```
const getBlocks = pathOr([], ['content', 'blocks']);
const setBlocks = assocPath(['content', 'blocks']);

const insertExperimentalBlock = options => pageData => {
  if (!options.shouldInsert) return pageData;

  return setBlocks(
    insert(options.insertAtIndex, options.blockModel, getBlocks(pageData)),
    pageData,
  );
};
```

This could be a generic-ish function whereby you supply it with an object of options. Should it be inserted, where, and the actual block content. We already have a couple of functions that do this:
- https://github.com/bbc/simorgh/blob/aba81751518d34516a75cc285c749ab80ba2c98e/src/app/routes/cpsAsset/getInitialData/addRecommendationsBlock/index.js
- https://github.com/bbc/simorgh/blob/5d950b942b118948420e3405e718c6a9ddba21bc/src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.js

This could work, but there would still be a lot of custom logic within the components we want in the experiment, so I'm not certain of the re-usability.

**Comment by amoore108** [comment]
Just to note, I also tested the above on a page of includes whilst using the `useOptimizelyVariation` hook and it didn't break the includes. This is not to say the above method is the only way to not break includes, but it seems like modifying data in `getInitialData` doesn't affect include re-rendering.

I did a very hacky check to split up the recommendations for the split-variation:
```
<RecommendationsPromoList
            promoItems={
              promoVariation === 'variation_1'
                ? index === 0
                  ? items.slice(0, 2)
                  : items.slice(2, 4)
                : items
            }
/>
```

It renders the recs as normal (full list of 4), then when the Optimizely hook completes and returns the variation ID, it splits them. The includes render and behave as normal.

Similarly if I try something like:
```
  if (promoVariation === 'variation_1') return null;
```

This will return nothing for variation ID `variation_1`. The recs display as normal (full list of 4) and then disappear once the variation ID is resolved. The includes render and behave as normal again.
